SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -933,33 +933,38 @@ protected void unbindAndShutdown() {
         }
     }
 
-    protected void removeCurrentDialog() {
+    protected void removeCurrentDialog(Activity callerActivity) {
```

Comment:
I think we only need to remove the dialog if the owner activity of the dialog is finishing. You can find this out by calling dialog.getOwnerActivity().isFinishing(). If its nots finishing then don't remove the dialog. This way we don't need to add Activity to the lifecyle methods.